Signal/src/ViewControllers/OWSConversationSettingsTableViewController.m

```diff
@@ -115,6 +120,58 @@ - (BOOL)isGroupThread
 - (void)configureWithThread:(TSThread *)thread
 {
     self.thread = thread;
+    [self updateEditButton];
+}
+
+- (void)updateEditButton
+{
+    OWSAssert(self.thread);
+
+    if ([self.thread isKindOfClass:[TSContactThread class]]) {
+        self.navigationItem.rightBarButtonItem =
+            [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"EDIT_TXT", nil)
+                                             style:UIBarButtonItemStylePlain
+                                            target:self
+                                            action:@selector(didTapEditButton)];
+    }
+}
+
+- (void)didTapEditButton
+{
+    if (![self.thread isKindOfClass:[TSContactThread class]]) {
+        DDLogError(@"%@ unexpected thread: %@ in %s", self.tag, self.thread, __PRETTY_FUNCTION__);
+        OWSAssert(NO);
+        return;
+    }
+
+    TSContactThread *contactThread = (TSContactThread *)self.thread;
+    [self.contactsViewHelper presentContactViewControllerForRecipientId:contactThread.contactIdentifier
+                                                     fromViewController:self
+                                                        editImmediately:YES];
```

Comment:
I'd rather add an explicit "Edit" right bar button to the group settings and remove the "tap-to-edit" functionality from the group settings. 

Motivation:
1. the right bar button "Edit" item is standard - users know to look for it.
2. tap to edit seems more likley to be misfired.

Separately, what do you think about using a "fade" transition for the "group edit" modal instead of the modal pop-over?